pkg/reconciler/v1alpha1/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -62,6 +62,23 @@ type ResolvedPipelineRunTask struct {
 // state of the PipelineRun.
 type PipelineRunState []*ResolvedPipelineRunTask
 
+func (state PipelineRunState) IsDone() (isDone bool) {
+	isDone = true
+	for _, t := range state {
+		if t.TaskRun == nil || t.PipelineTask == nil {
+			return false
+		}
+		status := t.TaskRun.Status.GetCondition(apis.ConditionSucceeded)
+		retriesDone := len(t.TaskRun.Status.RetriesStatus)
+		retries := t.PipelineTask.Retries
+		isDone = isDone && (status.IsTrue() || status.IsFalse() && retriesDone > retries)
```

Comment:
I'm a bit confused about `retriesDone` - how does the `RetriesStatus` slice become longer than `retries`?  Can `addRetryHistory` be called more than `retries` times?

The reason I ask is that I presumed that if I set `retries` to 0 then I would expect `RetriesStatus` to have a length of 0.  If I set `retries` to 1 I would expect `RetriesStatus` to have at most 1 entry.  I may have misunderstood the logic here though.